From 3c440102b1888424f704914609ee054453a2d866 Mon Sep 17 00:00:00 2001
From: Chaoting Liu <brli@chakralinux.org>
Date: Tue, 27 Aug 2019 11:37:15 +0200
Subject: [PATCH] include linux/sockios.h

---
 src/plugins/canbus/socketcan/socketcanbackend.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/plugins/canbus/socketcan/socketcanbackend.cpp b/src/plugins/canbus/socketcan/socketcanbackend.cpp
index 74b0d1d..a2da146 100644
--- a/src/plugins/canbus/socketcan/socketcanbackend.cpp
+++ b/src/plugins/canbus/socketcan/socketcanbackend.cpp
@@ -45,6 +45,7 @@
 
 #include <linux/can/error.h>
 #include <linux/can/raw.h>
+#include <linux/sockios.h>
 #include <errno.h>
 #include <unistd.h>
 #include <net/if.h>
-- 
2.20.1

